pylearn2/datasets/dataset.py

```diff
@@ -110,6 +111,88 @@ def iterator(self, mode=None, batch_size=None, num_batches=None,
         # can be handled here.
         raise NotImplementedError()
 
+    def _init_iterator(self, mode=None, batch_size=None, num_batches=None,
+                       rng=None, data_specs=None):
+        """
+        Return an iterator for this dataset with the specified
+        behaviour. Unspecified values are filled-in by the default.
```

Comment:
It is not returning an iterator, it is preparing the arguments to pass to the `iterator` method.